app/src/main/java/io/homeassistant/companion/android/sensors/NetworkSensorManager.kt

```diff
@@ -174,7 +176,25 @@ class NetworkSensorManager : SensorManager {
             conInfo = wifiManager.connectionInfo
         }
 
-        val bssid = if (conInfo!!.bssid == null) "<not connected>" else conInfo.bssid
+        var bssid = if (conInfo!!.bssid == null) "<not connected>" else conInfo.bssid
+
+        val settingName = "replace_$bssid"
+        val sensorDao = AppDatabase.getInstance(context).sensorDao()
+        val fullSensor = sensorDao.getSettings(bssidState.id)
+        val getNextBSSID = fullSensor?.settings?.firstOrNull { it.name == "get_next_bssid" }?.value ?: "false"
+        val currentSetting = fullSensor?.settings?.firstOrNull { it.name == settingName }?.value ?: ""
+        if (getNextBSSID == "true") {
+            if (currentSetting == "") {
+                sensorDao.add(Setting(bssidState.id, "get_next_bssid", "false", "toggle"))
+                sensorDao.add(Setting(bssidState.id, settingName, "", "string"))
```

Comment:
Default to the non original value?

Reply:
I kept it this way in case the user took too long on the settings page due to the refresh timing.  This way to them it all looks the same until they set the name they choose.

Reply:
I was suggesting this:
```suggestion
                sensorDao.add(Setting(bssidState.id, settingName, bssid, "string"))
```
This way the setting gets added but it will still report the same value to Home Assistant and not instantly delete ;-)